source/components/reusable/TextField/_tests/TextFieldContainer.test.js

```diff
@@ -1,37 +1,141 @@
+    makeEditableSpy = sinon.spy(TextFieldContainer.prototype, 'makeEditable')
+    makeKeyPressProjectSpy = sinon.spy(globalState, 'updateProjectText')
+    makeKeyPressCouldDoSpy = sinon.spy(globalState, 'updateCouldDoText')
+    wrapper = mount( < TextFieldContainer / > )
+    globalState.set(mockGlobalState)
   })
 
-  afterEach( () => {
+  afterEach(() => {
     warnStub.restore()
     makeEditableSpy.restore()
+    makeKeyPressProjectSpy.restore()
+    makeKeyPressCouldDoSpy.restore()
     wrapper.unmount()
+    moxios.uninstall()
   })
 
-  it( 'renders .text-field-container', () =>
-    expect( wrapper.find( '.text-field-container' ).length ).to.equal( 1 )
+  it('renders .text-field-container', () =>
+    expect(wrapper.find('.text-field-container').length).to.equal(1)
   )
 
-  it( 'calls makeEditable on click', () => {
-    wrapper.find( '.text-field-container' ).simulate( 'click' )
+  it('calls makeEditable on click', () => {
+    wrapper.find('.text-field-container').simulate('click')
     expect( makeEditableSpy.calledOnce ).to.equal( true )
   })
 
   it( 'renders <TextField />', () =>
-    expect( wrapper.find( 'TextField' ).length ).to.equal( 1 )
+    expect(wrapper.find('TextField').length).to.equal( 1 )
   )
 
-  // Write test for handleKeyPress
+  context( 'Tests for handleKeyPress', () => {
+
+    it( 'updates global state with new project when type is project', done => {
+      wrapper = mount( < TextFieldContainer type='project' / > )
+      wrapper.instance().makeEditable()
+      let event = {
+        key: 'Enter'
+      }
+      wrapper.instance().handleKeyPress( event )
+      moxios.wait( () => {
+        const request = moxios.requests.mostRecent()
+        request.respondWith({
+          status: 200,
+          response: ''
+        }).then( () => {
+          expect( makeKeyPressProjectSpy.calledOnce ).to.equal( true )
+          expect( makeKeyPressCouldDoSpy.notCalled ).to.equal( true )
+          done()
+        }).catch( done )
+      })
+    })
+
+    it( 'updates global state with new couldDo when type is couldDo', done => {
+      wrapper = mount( < TextFieldContainer type='could-do' / > )
+      wrapper.instance().makeEditable()
+      let event = {
+        key: 'Enter'
+      }
+      wrapper.instance().handleKeyPress( event )
+      moxios.wait(() => {
+        const request = moxios.requests.mostRecent()
+        request.respondWith({
+          status: 200,
+          response: ''
+        }).then(() => {
+          expect( makeKeyPressCouldDoSpy.calledOnce ).to.equal( true )
+          expect( makeKeyPressProjectSpy.notCalled ).to.equal( true )
+          done()
+        }).catch( done )
+      })
+    })
+
+    it( 'keyHandlePress sets state to false after enter', done => {
+      wrapper = mount( < TextFieldContainer type='could-do' projectId='2' id='2' text='Sally Moos' / > )
+      wrapper.instance().makeEditable()
+      let event = {
```

Comment:
Event is declared three times, always the same. Move it to under the describe and remove the duplicate declarations.